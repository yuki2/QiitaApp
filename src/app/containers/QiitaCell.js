// @flow
import React, { Component } from 'react';
import { StyleSheet, View, Text, Image, TouchableHighlight } from 'react-native';

import type { QiitaItem } from '../flow-type';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    marginLeft: 8,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 4,
    marginBottom: 4,
  },
  rightContainer: {
    flex: 1,
  },
  title: {
    fontSize: 15,
    textAlign: 'left',
    marginTop: 4,
    marginBottom: 4,
  },
  name: {
    fontSize: 12,
    marginLeft: 4,
    textAlign: 'left',
    color: 'gray',
  },
  tag: {
    fontSize: 12,
    textAlign: 'left',
    color: 'gray',
    marginTop: 4,
    marginBottom: 4,
  },
  thumbnail: {
    width: 12,
    height: 12,
  },
});

type Props = {
  item: QiitaItem,
  onSelect: (item: QiitaItem) => void,
};
export default class QiitaCell extends Component<Props> {
  static defaultProps = {
    onSelect: () => {},
    item: {},
  };

  createTag = (item: QiitaItem): string => item.tags.map(tag => tag.name).join(', ');

  render() {
    const { onSelect, item } = this.props;
    return (
      <TouchableHighlight onPress={onSelect}>
        <View style={styles.container}>
          <View style={styles.row}>
            <Image source={{ uri: item.user.profileImageUrl }} style={styles.thumbnail} />
            <Text style={styles.name}>{item.user.id}</Text>
          </View>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.tag}>{item.tags.map(tag => tag.name).join(', ')}</Text>
        </View>
      </TouchableHighlight>
    );
  }
}
