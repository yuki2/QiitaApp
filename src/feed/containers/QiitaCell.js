import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { StyleSheet, View, Text, Image, TouchableHighlight } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    marginLeft: 8,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 4,
    marginBottom: 4,
  },
  rightContainer: {
    flex: 1,
  },
  title: {
    fontSize: 15,
    textAlign: 'left',
    marginTop: 4,
    marginBottom: 4,
  },
  name: {
    fontSize: 12,
    marginLeft: 4,
    textAlign: 'left',
    color: 'gray',
  },
  tag: {
    fontSize: 12,
    textAlign: 'left',
    color: 'gray',
    marginTop: 4,
    marginBottom: 4,
  },
  thumbnail: {
    width: 12,
    height: 12,
  },
});

export default class QiitaCell extends Component {
  static defaultProps = {
    onSelect: () => {},
    item: {},
  };
  constructor(props) {
    super(props);
    this.onSelect = this.props.onSelect;
    this.item = this.props.item;
  }

  createTag = (item) => {
    item.tags.map(tag => tag.name).join(', ');
  };

  render() {
    return (
      <TouchableHighlight onPress={() => this.onSelect()}>
        <View style={styles.container}>
          <View style={styles.row}>
            <Image source={{ uri: this.item.user.profile_image_url }} style={styles.thumbnail} />
            <Text style={styles.name}>{this.item.user.id}</Text>
          </View>
          <Text style={styles.title}>{this.item.title}</Text>
          <Text style={styles.tag}>{this.item.tags.map(tag => tag.name).join(', ')}</Text>
        </View>
      </TouchableHighlight>
    );
  }
}

QiitaCell.propTypes = {
  onSelect: PropTypes.func,
  item: PropTypes.object,
};
