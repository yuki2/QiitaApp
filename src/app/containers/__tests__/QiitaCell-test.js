import React from 'react';
import renderer from 'react-test-renderer';
import QiitaCell from '../QiitaCell';

const user = {
  description: 'description',
  facebookId: 'facebookId',
  followeesCount: 1,
  followersCount: 1,
  githubLoginName: 'githubLoginName',
  id: 'id',
  itemsCount: 1,
  name: 'name',
  organization: 'organization',
  permanentId: 'permanentId',
  profileImageUrl: 'profileImageUrl',
  twitterScreenName: 'twitterScreenName',
  websiteUrl: 'websiteUrl',
};

const item = {
  id: 'id',
  title: 'title',
  url: 'url',
  user,
  tags: ['tag1', 'tag2'],
  likesCount: 1,
};

it('renders correctly', () => {
  const onSelectItem = () => {};
  const tree = renderer.create(<QiitaCell onSelect={onSelectItem} item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});
