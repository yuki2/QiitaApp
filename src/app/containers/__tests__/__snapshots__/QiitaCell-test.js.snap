// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 4,
          "marginTop": 4,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "profileImageUrl",
          }
        }
        style={
          Object {
            "height": 12,
            "width": 12,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 12,
            "marginLeft": 4,
            "textAlign": "left",
          }
        }
      >
        id
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "flex": 1,
            "fontSize": 12,
            "marginLeft": 4,
            "textAlign": "right",
          }
        }
      >
        a few seconds ago
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 15,
          "marginBottom": 4,
          "marginTop": 4,
          "textAlign": "left",
        }
      }
    >
      title
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 4,
          "marginTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 4,
            "marginTop": 4,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gray",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "gray",
                "fontSize": 12,
                "marginLeft": 4,
                "textAlign": "left",
              },
              Object {
                "marginRight": 8,
              },
            ]
          }
        >
          1
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "gray",
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 12,
            "marginLeft": 4,
            "textAlign": "left",
          }
        }
      >
        , 
      </Text>
    </View>
  </View>
</View>
`;
